flexmeasures/data/schemas/scheduling/__init__.py

```diff
@@ -245,14 +246,18 @@ def forbid_time_series_specs(self, data: dict, **kwargs):
     @validates_schema
     def forbid_fixed_prices(self, data: dict, **kwargs):
         """Do not allow fixed consumption price or fixed production price in the flex-context fields saved in the db."""
-        if "consumption_price" in data and isinstance(data["consumption_price"], str):
+        if "consumption_price" in data and isinstance(
+            data["consumption_price"], pint.Quantity
+        ):
             raise ValidationError(
-                "Fixed prices are not currently supported in flex-context fields in the DB."
+                "Fixed prices are not currently supported for consumption_price in flex-context fields in the DB."
```

Comment:
The user facing validation error should refer to the field name as the user experiences it, which is with dashes: `consumption-price`.